src/iuse.cpp

```diff
@@ -4255,6 +4255,39 @@ int iuse::hand_crank( player *p, item *it, bool, const tripoint & )
     return 0;
 }
 
+int iuse::hand_crank_air( player *p, item *it, bool, const tripoint & )
+{
+    if( p->is_npc() ) {
+        // Long action
+        return 0;
+    }
+    if( p->is_underwater() ) {
+        p->add_msg_if_player( m_info, _( "It's not waterproof enough to work underwater." ) );
+        return 0;
+    }
+    if( p->get_fatigue() >= DEAD_TIRED ) {
+        p->add_msg_if_player( m_info, _( "You're too exhausted to keep cranking." ) );
+        return 0;
+    }
+    item *magazine = it->magazine_current();
+    if( magazine && magazine->has_flag( "RECHARGE_AIR" ) ) {
+        int time = to_moves<int>( 1600_minutes );
+        if( it->ammo_capacity() > it->ammo_remaining() ) {
+            p->add_msg_if_player( string_format( _( "You start cranking the %s to charge its %s." ),
```

Comment:
`add_msg_if_player` does it's own string formatting, you don't need to do it yourself.